lib/chef/resource/windows_pagefile.rb

```diff
@@ -118,15 +135,25 @@ def validate_name
           raise "#{new_resource.path} does not match the format DRIVE:\\path\\file.sys for pagefiles. Example: C:\\pagefile.sys"
         end
 
+        # The powershell cmdlets to create a new pagefile will happily report success when the drive you just created the pagefile for does not exist.
+        def pagefile_drive_exist?(pagefile)
+          root_drive = pagefile.split(":")[0]
```

Comment:
You can probably simplify this whole method a bit:

```ruby
def pagefile_drive_exist?(pagefile)
  ::Dir.exist?(pagefile[0] + ":\\")
end
```

This is because

```
irb(main):003:0> pagefile[0]
=> "C"
irb(main):004:0> pagefile[0] + ":\\"
=> "C:\\"
```

Dir.exist is going to return a truth value anyways so there's no need to modify it